### PR TITLE
Implement LSP symbol rename

### PR DESCRIPTION
**Pull request by nexus6-haiku**

### CODE REVIEW

**Comment by Haibadjba** [comment]
Hello

File: src/lsp-client/LSPEditorWrapper.cpp [hunk 1]
```diff
@@ -10,6 +10,7 @@
 #include <Path.h>
 #include <Window.h>
 #include <Catalog.h>
+#include <Json.h>
```
Comment (Freaxed) [inline src/lsp-client/LSPEditorWrapper.cpp:13]: Is this needed?

File: src/ui/Editor.h [hunk 2]
```diff
@@ -13,6 +13,7 @@
 #include <ScintillaView.h>
 
 #include "LSPCapabilities.h"
+#include "json_fwd.hpp"
```
Comment (Freaxed) [inline src/ui/Editor.h:16]: Is this needed?

File: src/ui/GenioWindow.cpp [hunk 3]
```diff
@@ -54,6 +54,7 @@
 #include "GoToLineWindow.h"
 #include "GSettings.h"
 #include "IconMenuItem.h"
+#include "json.hpp"
```
Comment (Freaxed) [inline src/ui/GenioWindow.cpp:57]: this include is huge.. do we really need it? It looks like the change is only based on strings

File: src/lsp-client/LSPEditorWrapper.cpp [hunk 4]
```diff
@@ -210,6 +227,23 @@ LSPEditorWrapper::_DoFormat(json& params)
 }
 
 
+void
+LSPEditorWrapper::_DoRename(json& params)
+{
+	printf("_DoRename: %s\n", params.dump().c_str());
```
Comment (Freaxed) [inline src/lsp-client/LSPEditorWrapper.cpp:233]: please remove this printf

File: src/lsp-client/LSPEditorWrapper.cpp (hunk 1 above)
Comment (nexus6-haiku) [reply src/lsp-client/LSPEditorWrapper.cpp]: done

File: src/ui/Editor.h (hunk 2 above)
Comment (nexus6-haiku) [reply src/ui/Editor.h]: done

File: src/lsp-client/LSPEditorWrapper.cpp (hunk 4 above)
Comment (nexus6-haiku) [reply src/lsp-client/LSPEditorWrapper.cpp]: done

File: src/ui/GenioWindow.cpp (hunk 3 above)
Comment (nexus6-haiku) [reply src/ui/GenioWindow.cpp:57]: done

**Review by jackburton79 — APPROVED** [review]

**Pull request merged**